src/main/target/NUCLEOH743/target.h

```diff
@@ -0,0 +1,250 @@
+
+#define USE_I2C
+#define USE_I2C_DEVICE_1
+#define I2C1_SCL                PB8
+#define I2C1_SDA                PB9
+#define I2C_DEVICE (I2CDEV_1)
+
+#define USE_MAG
+#define USE_MAG_HMC5883
+#define USE_MAG_SPI_HMC5883
+#define HMC5883_SPI_INSTANCE    NULL
+#define HMC5883_CS_PIN          NONE
+
+#define USE_BARO
+#define USE_BARO_BMP085
+#define USE_BARO_BMP280
+#define USE_BARO_MS5611
+#define USE_BARO_SPI_BMP280
+#define BMP280_SPI_INSTANCE     NULL
+#define BMP280_CS_PIN           NONE
+
+#define USE_GYRO
+#define USE_MULTI_GYRO
+#define USE_ACC
+
+#define USE_GYRO_SPI_MPU6000
+#define USE_ACC_SPI_MPU6000
+#define USE_GYRO_SPI_MPU6500
+#define USE_ACC_SPI_MPU6500
+#define USE_GYRO_SPI_MPU9250
+#define USE_ACC_SPI_MPU9250
+
+#define GYRO_1_CS_PIN           PD15
+#define GYRO_1_SPI_INSTANCE     SPI1
+
+// I2C acc/gyro test, may require to activate
+// set gyro_x_bustype = I2C
+// set gyro_x_i2cBus = <Bus ordinal of I2C_DEVICE>
+//#define USE_GYRO_MPU6050
+//#define USE_ACC_MPU6050
+
+#define USE_FLASH_CHIP
+#define USE_FLASH_M25P16
+#define USE_FLASH_W25M
+#define FLASH_SPI_INSTANCE      NULL
+#define FLASH_CS_PIN            NONE
+#define USE_FLASHFS
+
+#define USE_BRUSHED_ESC_AUTODETECT  // Detect if brushed motors are connected and set defaults appropriately to avoid motors spinning on boot
+#define USE_GYRO_REGISTER_DUMP  // Adds gyroregisters command to cli to dump configured register values
+#define USE_TIMER
+#define USE_PWM_OUTPUT
+#define USE_MOTOR
+#define USE_EXTI
+
+#define USE_RANGEFINDER
+#define USE_RANGEFINDER_HCSR04
+#define USE_RANGEFINDER_TF
+
+#define USE_TRANSPONDER
+
+#define USE_MAX7456
+#define MAX7456_SPI_INSTANCE    NULL // SPI3
+#define MAX7456_SPI_CS_PIN      NONE // PC9
+#define MAX7456_SPI_CLK         (SPI_CLOCK_STANDARD)
+
+#define USE_I2C_OLED_DISPLAY
+
+#define USE_DMA
+
+// H7 allows all stream to handle any peripheral
+// No need for channel definition; each peripheral i/o has a unique channel thanks to MUX.
+//#define UART1_TX_DMA_STREAM     DMA1_Stream0
+//#define UART2_TX_DMA_STREAM     DMA1_Stream1
+//#define UART3_TX_DMA_STREAM     DMA1_Stream2
+//#define UART4_TX_DMA_STREAM     DMA1_Stream3
+//#define UART5_TX_DMA_STREAM     DMA1_Stream4
+//#define UART6_TX_DMA_STREAM     DMA1_Stream5
+//#define UART7_TX_DMA_STREAM     DMA1_Stream6
+//#define UART8_TX_DMA_STREAM     DMA1_Stream7
+
+#define ADC1_INSTANCE ADC1
+#define ADC2_INSTANCE ADC2
+#define ADC3_INSTANCE ADC3
+#define ADC1_DMA_STREAM DMA2_Stream0
+#define ADC2_DMA_STREAM DMA2_Stream1
+#define ADC3_DMA_STREAM DMA2_Stream2
+
+#define VBAT_ADC_PIN            PB1  // ADC1
+#define CURRENT_METER_ADC_PIN   PC0  // ADC1
+#define RSSI_ADC_PIN            PF14 // ADC2
+#define EXTERNAL1_ADC_PIN       PC3  // ADC3
+
+#define DEFAULT_VOLTAGE_METER_SOURCE VOLTAGE_METER_ADC
+#define DEFAULT_CURRENT_METER_SOURCE CURRENT_METER_ADC
+
+#define USE_DSHOT
+#define USE_DSHOT_DMAR
+
+#undef USE_USB_MSC
+#undef USE_OVERCLOCK
```

Comment:
Are these generically not supported in H7 at the moment? If so, we should undefine / not define them in `common_pre.h` for H7, to avoid having to duplicate this on every H7 target.